packages/fury-adapter-oas3-parser/test/unit/parser/oas/parseParameterObject-test.js

```diff
@@ -62,6 +61,78 @@ describe('Parameter Object', () => {
       expect(parseResult.get(0)).to.be.instanceof(namespace.elements.Member);
       expect(parseResult.get(0).key.toValue()).to.equal(unreserved);
     });
+
+    it('auto pct-encode reserved chars in query name', () => {
+      const reserved = '!*\'();:@&=+$,?%#[]';
+      const encoded = '%21%2A%27%28%29%3B%3A%40%26%3D%2B%24%2C%3F%25%23%5B%5D';
+
+      const parameter = new namespace.elements.Object({
+        name: reserved,
+        in: 'query',
+      });
+
+      const parseResult = parse(context, parameter);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.Member);
+      expect(parseResult.get(0).key.toValue()).to.equal(encoded);
+    });
+
+    it('avoid double encoding pct-encode in query name', () => {
+      const input = 'user%5bname%5d';
+
+      const parameter = new namespace.elements.Object({
+        name: input,
+        in: 'query',
+      });
+
+      const parseResult = parse(context, parameter);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.Member);
+      expect(parseResult.get(0).key.toValue()).to.equal(input);
+    });
+
+    it('allows header name to contain unreserved characters', () => {
+      const alpha = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+      const digit = '0123456789';
+      const unreserved = `${alpha}${digit}-._~`;
+
+      const parameter = new namespace.elements.Object({
+        name: unreserved,
+        in: 'header',
+      });
+
+      const parseResult = parse(context, parameter);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult.get(0)).to.be.instanceof(namespace.elements.Member);
+      expect(parseResult.get(0).key.toValue()).to.equal(unreserved);
+    });
+
+    it('does not allow header name to contain reserved keyword `Accept`', () => {
+      const parameter = new namespace.elements.Object({
+        name: 'accept',
+        in: 'header',
+      });
+
+      const parseResult = parse(context, parameter);
+
+      expect(parseResult.length).to.equal(1);
+      expect(parseResult).to.contain.warning('\'Parameter Object\' \'name\' in location \'header\' ignore keywords \'Accept\', \'Content-Type\' and \'Authorization\'');
+    });
+
+    it('does not allow header name to contain reserved keyword `Content-Type`', () => {
```

Comment:
We should have one more case of this type of test for `Authorization`.
```suggestion
    it('does not allow header name to contain reserved keyword `Authorization`', () => {
      const parameter = new namespace.elements.Object({
        name: 'Authorization',
        in: 'header',
      });

      const parseResult = parse(context, parameter);

      expect(parseResult.length).to.equal(1);
      expect(parseResult).to.contain.warning('\'Parameter Object\' \'name\' in location \'header\' ignore keywords \'Accept\', \'Content-Type\' and \'Authorization\'');
    });
 
    it('does not allow header name to contain reserved keyword `Content-Type`', () => {
```